engine/src/main/java/org/camunda/bpm/engine/impl/cmd/OverwriteHistoryLevelCmd.java

```diff
@@ -0,0 +1,43 @@
+package org.camunda.bpm.engine.impl.cmd;
+
+import org.camunda.bpm.engine.impl.SchemaOperationsProcessEngineBuild;
+import org.camunda.bpm.engine.impl.cfg.ProcessEngineConfigurationImpl;
+import org.camunda.bpm.engine.impl.db.entitymanager.DbEntityManager;
+import org.camunda.bpm.engine.impl.history.HistoryLevel;
+import org.camunda.bpm.engine.impl.interceptor.Command;
+import org.camunda.bpm.engine.impl.interceptor.CommandContext;
+
+
+public class OverwriteHistoryLevelCmd implements Command<Void> {
```

Comment:
After a good night sleep, I believe calling this "DetermineHistoryLevelCmd" and letting it return a history level would be more intuitive. We would still have to pass the list of available historyLevels, in case the use defined custom ones.